base/loading.jl

```diff
@@ -526,6 +526,15 @@ evaluated by `julia -e <expr>`.
 """
 macro __DIR__() source_dir() end
 
+"""
+    current_location() -> Int
+
+Return the line number where the current macro expansion was invoked.
+"""
+function current_location()
+    convert(Int, ccall(:jl_current_lineno, Cint, ()))
```

Comment:
No, but good question. `ptls->current_lineno` is not really a replacement for `jl_lineno`, but rather a place to stash the line number of the macro caller.  (The only place it's set is inside `fl_invoke_julia_macro`.)  As such I've probably named it badly, what if I rename it as `macro_caller_lineno` or something?

Side question - is there an approved way to load thread local state from julia without creating a new C function specifically for the purpose?  I didn't see one.